apps/backend/src/app/api/latest/auth/oauth/callback/[provider_id]/route.tsx

```diff
@@ -284,54 +305,50 @@ const handler = createSmartRouteHandler({
 
                     // Check if we should link this OAuth account to an existing user based on email
                     if (oldContactChannel && oldContactChannel.usedForAuth) {
-                      const mergeOAuthMethods = tenancy.config.merge_oauth_methods.toUpperCase();
+                      const oauthAccountMergeStrategy = tenancy.config.oauth_account_merge_strategy.toUpperCase();
+                      switch (oauthAccountMergeStrategy) {
+                        case 'LINK_METHOD': {
+                          if (!oldContactChannel.isVerified) {
+                            throw new KnownErrors.UserEmailAlreadyExists();
+                          }
 
-                      if (mergeOAuthMethods === 'LINK_METHOD') {
-                        // Link this OAuth account to the existing user instead of creating a new one
-                        const existingUser = oldContactChannel.projectUser;
+                          if (!userInfo.emailVerified) {
+                            throw new KnownErrors.EmailNotVerified();
```

Comment:
This should be more descriptive (the current error message says very little for the user). We should probably also call `captureError`, so it turns up on Sentry.

In fact, it's probably better here to throw the same error that RAISE_ERROR throws instead of EmailNotVerified. That one explains for the user and tells them exactly what they have to do (ie. login with their other provider).